### DISCUSSION TITLE
jquery ajax call post method with json return http 401 UNAUTHORIZED

### DISCUSSION THREAD

**Discussion by easykoala** [tokio-rs/axum#1026] [Q&A] [answered]
## Bug Report

jquery ajax call post method with json return http 401 UNAUTHORIZED
but postman tested ok

### Version

axum 0.5.5

### Platform

Windows 10

**Comment by davidpdrsn** [comment]
Please provide more details.

**Reply by easykoala** [reply]
When remove below code, it run ok. why:

            .layer(
                CorsLayer::new()
                    .allow_origin(Origin::exact("http://localhost:3000".parse().unwrap()))
                    .allow_methods(vec![Method::GET, Method::POST]),
            )

but postman run always ok.

ajax calling:

				ajaxPro({url: '/api/device/siteUpdate', data: params}, function(){
					......
				});

function ajaxPro(ajax, callback) {
	var baseAjax = {
		"type" : 'POST',
		"dataType" : "json", 
	}

	$.ajax($.extend(baseAjax, ajax)).done(function(json) {
		if (typeof callback === 'function') {
		    if(json.code==0)
			    callback(json);
			else
			    alert(json.msg)
		}
	}).fail(function(xhr, textStatus, errorThrown) {
		alert(xhr.status + ' ' + textStatus + " " + errorThrown);
	});
}

**Reply by easykoala** [reply]
get method has no effect.

**Reply by davidpdrsn** [reply]
My guess is that postman doesn't actually check the CORS policy on the response and that only the browser does that.

What error do you see?

**Comment by jplatte** [comment, marked as answer]
It sounds like you're using an old version of tower-http. As of version 0.3.0, the `CorsLayer` won't ever return 401s. I would recommend upgrading.

**Reply by easykoala** [reply]
You are great！the tower-http of version is 0.2，it's copy from example's. hahah
It's OK, Thank you.

**Reply by jplatte** [reply]
Not sure what you mean since the `cors` example uses tower-http 0.3.0, but good to hear your problem is solved 🙂
